sql/connect/server/src/test/scala/org/apache/spark/sql/connect/ProtoToParsedPlanTestSuite.scala

```diff
@@ -259,4 +261,35 @@ class ProtoToParsedPlanTestSuite
       writer.close()
     }
   }
+
+  protected def normalizeRelation(rel: proto.Relation): proto.Relation = {
+    rel.getRelTypeCase match {
+      case proto.Relation.RelTypeCase.READ =>
+        proto.Relation.newBuilder(rel).setRead(normalizeDataSource(rel.getRead)).build()
+      case _ => rel
+    }
+  }
+
```

Comment:
The following logic is referenced here:
https://github.com/apache/spark/blob/b72bf79f5a26f1a57527fbb083bd3c74d42169e8/sql/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala#L1366-L1377